core/domain/user_domain.py

```diff
@@ -345,3 +348,23 @@ def validate(self):
             raise utils.ValidationError(
                 'Expected can_review_questions to be a boolean value, '
                 'found: %s' % type(self.can_review_questions))
+
+
+class ModifiableUserData(python_utils.OBJECT):
+    """Domain object for the ModifiableUserData."""
+
+    def __init__(
+            self, display_alias, last_agreed_to_terms, last_logged_in, user_bio,
```

Comment:
This needs to be restricted to what is modifiable. You have too much stuff here. Keep it to the bare minimum for now (display_alias and pin, I believe). Maybe language code preferences if those are configurable per-profile in Android (not sure, you'll have to check). And the user ID too (add a comment above it saying that this is not intended to be modifiable, it just identifies the object). 

In addition, it occurs to me that we need some plan for what to do here when the schema for this object (and corresponding client payloads) changes in the future. I think maybe name the class ModifiableUserDataV1 perhaps, and then when the Android client sends info we can read the schema version and figure out which class to use appropriately. That should work for now, but perhaps start a discussion with @BenHenning and @vinitamurthi about this.

Reply:
Yes, the language code preferences are configurable per profile. The attributes `user_id`, `display_alias`, `pin`, all kinds of `language codes - will be kept. I think we should keep subject interests also (but I haven't worked on understanding how exactly it will be used on Android app) - but it's not strictly a requirement to "support" profile itself so I can drop it for now.
So, okay I think your advice of adding things later on makes sense to me. 

@vinitamurthi what are your views on this? Particularly would only above 4 kinds of attribute work or I can add subject interests too if that feature is completely ready for use.

Reply:
FYI I'm pretty sure subject interests aren't used in Android atm

Reply:
Changes:
- Updated the class name to `ModifiableUserDataV1`.
- Attributes: `user_id`, `display_alias`, `pin`, languages code related attributes

@seanlip PTAL if the naming and its usage is what you wanted for now.
@vinitamurthi @BenHenning Regarding plan for updating its schema in the future, what can I do exactly - create an issue mentioning what attributes have to be added in future or should we talk about it separately?